04/SpeedTest.js

```diff
@@ -41,37 +65,42 @@ const SpeedTest = () => {
     };
 
     useEffect(() => {
-        if (word && text) {
-            const counted = commonSameCount(word, text);
-            console.log(counted);
+        if (generatedWord && text) {
+            const counted = commonSameCount(generatedWord, text);
             setSigns(counted);
         }
-        if (text === word) {
-            setText('');
+        if (text === generatedWord) {
+            setArray(oldArray => [...oldArray, taskTime]);
+
+            dispatch({ type: 'word' });
+            dispatch({ type: 'time' });
             clearInterval(intervalRef.current);
-            setArray(oldArray => [...oldArray, time]);
             setTime(0);
             regenerateWord();
-            timerFunc();
+            countTime();
         }
     }, [text]);
+
     useEffect(() => {
-        console.log(timeArray);
-    }, [timeArray]);
+        if (results.time !== false) {
+            setResArr(oldArray => [...oldArray, results]);
+        }
+    }, [results.time]);
```

Comment:
Jeśli nie będziesz aktualizował tych danych w `useEffect()`, a w miesjcu gdzie faktycznie ma mieć to miejsce czyli wew. `if(text === generatedWord) { ... }` to wyniki powinny być prawidłowo pokazywane w tabeli.